cmd/controller/app/server.go

```diff
@@ -55,19 +59,24 @@ func NewControllerManagerCommand() *cobra.Command {
 		klog.Fatal("Failed to load configuration from disk", err)
 	}
 
+	// Initialize command to run our controllers later
 	cmd := &cobra.Command{
 		Use:   "controller-manager",
 		Short: `KubeSphere DevOps controller manager`,
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
 			if errs := s.Validate(); len(errs) == 0 {
-				err = run(s, signals.SetupSignalHandler())
+				if err = run(s, signals.SetupSignalHandler()); err != nil {
+					klog.Error("Failed to run controller manager", err)
```

Comment:
We don't need to print the error log here. The library cobra will print it via err. And cobra allows us to set the stdout and stderr. That means it can output a different place.